src/hocs/withContext.js

```diff
@@ -252,3 +252,256 @@ export function withContext(Component) {
+        ],
+    );
+
+    const itemsState = React.useMemo(
+        () =>
+            getItemsState({
+                config: props.config,
+                itemsStateAndParams: props.itemsStateAndParams,
+            }),
+        [props.config, props.itemsStateAndParams],
+    );
+
+    const getItemsMeta = React.useCallback((pluginsRefs) => {
+        return pluginsRefs
+            .map((ref) => {
+                if (!(ref && typeof ref.getMeta === 'function')) {
+                    return undefined;
+                }
+                return ref.getMeta();
+            })
+            .filter(Boolean);
+    }, []);
+
+    const getLayout = React.useCallback(() => {
+        return props.layout.map((item) => {
+            if (item.i in adjustedLayouts.current) {
+                return {
+                    ...adjustedLayouts.current[item.i],
+                    w: item.w,
+                    x: item.x,
+                    y: item.y,
+                };
+            } else {
+                return {...item};
+            }
+        });
+    }, [props.layout]);
+
+    const reloadItems = React.useCallback((pluginsRefs, data) => {
+        pluginsRefs.forEach((ref) => ref && ref.reload && ref.reload(data));
+    }, []);
+
+    return React.useMemo(
+        () => ({
+            getLayout,
+            config: props.config,
+            context: props.context,
+            overlayControls: props.overlayControls,
+            noOverlay: props.noOverlay,
+            defaultGlobalParams: props.globalParams,
+            globalParams: props.globalParams,
+            editMode: props.editMode,
+            settings: props.settings,
+            itemsState,
+            itemsParams,
+            registerManager: props.registerManager,
+            onItemStateAndParamsChange,
+            removeItem: onItemRemove,
+            editItem: props.onItemEdit,
+            layoutChange: onLayoutChange,
+            getItemsMeta,
+            reloadItems,
+            memorizeOriginalLayout,
+            revertToOriginalLayout,
+            forwardedMetaRef: props.forwardedMetaRef,
+            draggableHandleClassName: props.draggableHandleClassName,
+        }),
+        [
+            getLayout,
+            props.config,
+            props.context,
+            props.overlayControls,
+            props.noOverlay,
+            props.globalParams,
+            props.editMode,
+            props.settings,
+            itemsState,
+            itemsParams,
+            props.registerManager,
+            onItemStateAndParamsChange,
+            onItemRemove,
+            props.onItemEdit,
+            onLayoutChange,
+            getItemsMeta,
+            reloadItems,
+            memorizeOriginalLayout,
+            revertToOriginalLayout,
+            props.forwardedMetaRef,
+            props.draggableHandleClassName,
+        ],
+    );
+}
+
+export function withMemoContext(Component) {
+    const WithMemoContext = React.memo((props) => {
```

Comment:
`memo` without comparison function is seems redundant

Reply:
done